Review edit

br-ndt opened the pull request:
- add reviewPatch to AttractionShowPage
- add reviewsRouter.patch route
- add user authentication for reviewPatch

Commit:
add editReview function
add EditReviewForm
add patch request in backend

Co-authored-by: camziny <camziny@users.noreply.github.com>
Co-authored-by: Fabricius Eusebio <yoFabricius@users.noreply.github.com>
Co-authored-by: Tyler <br.ndt.tyler@gmail.com>

Commit:
Merge branch 'reviewDelete' into review_edit

Commit:
finish patchReview function
finish EditReviewForm
add error handling

br-ndt committed:
Merge remote-tracking branch 'origin' into review_edit

br-ndt committed:
Merge branch 'main' into review_edit

nwalberts commented on client/src/components/EditReviewForm.js:
```diff
@@ -0,0 +1,57 @@
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    props.patchReview(editReview, props.id, props.toggleEdit);
```
okay, this is rather clever passing toggleEdit in here, but the downside is it creates some weird coupling in your app. 

This is what I would do
- have `patchReview` return true or false
- Note: this means you may have to await `props.patchReview`, which means youll also have to make your handleSbumit async
- below line 20 in this component, if patchReview returns true, you'll toggleEdit, if not, you wont run toggle edit

br-ndt committed:
address @nwalberts feedback
- change toggleEdit callback
- fix user detection for reviews
- fix review patch content nullity not being persisted
